### PR TITLE
let forkChoice onTick fail

### PR DESCRIPTION
**Pull request by tbenr**
*** NEEDS TO BE CAREFULLY ANALYZED ***

If there are possible deadlock conditions, we should avoid "joining" forever and let timeout so we can allow deadlock to resolve. We should double-check and, if we fail, we remain in a recoverable state, not in a broken-corrupted state.

### CODE REVIEW

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java [hunk 1]
```diff
@@ -298,7 +301,11 @@ public void onTick(
       final UInt64 currentTimeMillis, final Optional<TickProcessingPerformance> performanceRecord) {
     final UpdatableStore store = recentChainData.getStore();
     final UInt64 slotAtStartOfTick = spec.getCurrentSlot(store);
-    tickProcessor.onTick(currentTimeMillis).join();
+    try {
+      tickProcessor.onTick(currentTimeMillis).get(30, TimeUnit.SECONDS);
+    } catch (final InterruptedException | TimeoutException | ExecutionException e) {
+      throw new RuntimeException(e);
+    }
     performanceRecord.ifPresent(TickProcessingPerformance::tickProcessorComplete);
```
Comment (rolfyone) [inline ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java:309]: i think it's worth doing this finish in an 'always' block

also, we should probably put in a counter for failed ticks...

**Comment by rolfyone** [comment]
potentially given the stacktrace we looked at, raised #8097 as an alternative

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java (hunk 1 above)
Comment (tbenr) [reply ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java:309]: yes, right

**Comment by rolfyone** [comment]
i think if we do this, we'd plumb it in so we can tweak the value, and we'd start it high. I like the idea that timedTick can timeout, but being able to alter that value if its too low sometimes would be much easier if we could do it with cli arg...

**Comment by rolfyone** [comment]
i think we can park this....

**Pull request closed without merging**